frame/alliance/src/lib.rs

```diff
@@ -705,6 +707,19 @@ pub mod pallet {
 		pub fn join_alliance(origin: OriginFor<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
+			// We don't want anyone to join as an Ally before the Founders/Fellows have been
+			// initialized via Root call. The reasons are two-fold:
+			//
+			// 1. There is no `Rule` or admission criteria, so the joiner would be an ally to
+			//    nought, and
+			// 2. If one joins as an Ally and then is initialized via Root, we'll have a condition
+			//    where a member is both an Ally and a Founder or Fellow, causing lookup issues on
+			//    calls like `retire` and `kick_member`.
+			ensure!(
```

Comment:
note that on `init_members` we have different condition to decide if members were initialized. [link](https://github.com/paritytech/substrate/blob/3c0dc95999d190b2f8edc9f6c1be0f606ff30a4f/frame/alliance/src/lib.rs#L624)
> if any member exist (including Ally) => Members Initialized 

here

> if Founders and Fellows exists => Alliance Initialized 

Probably `MembersInitialized` and `AllianceInitialized` means the same, but have different conditions.

Also I mentioned, init_members and build on genesis can be called with only list of Ally, and after it will be impossible to onboard any voting members.

We need to have a single condition for `is_alliance_initialized` nether has_founders && has_fellows, or has_founders. Use same condition everywhere we need to check if alliance initialized. And incapsulate the logic to the separate method (`is_alliance_initialized`).
And since the `init_members` supposed to be called only once, we need to make sure the passed set of members is enough to initialize the alliance.